### PR TITLE
[14.0][ADD] shopfloor_single_product_transfer_frontend

### PR DESCRIPTION
**Pull request by JuMiSanAr**
Note: This PR includes a hook that's needed to make sure that the endpoints are synced if the shopfloor application was installed prior to adding this scenario.

Depends on:

- https://github.com/OCA/wms/pull/483
- https://github.com/OCA/wms/pull/513

ref: rau-48

### CODE REVIEW

File: shopfloor_single_product_transfer_mobile/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2022 Camptocamp SA (http://www.camptocamp.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Shopfloor single product transfer mobile",
+    "summary": "Mobile frontend for single product transfer scenario",
+    "version": "14.0.1.0.0",
+    "development_status": "Beta",
+    "depends": ["shopfloor_mobile_base", "shopfloor_single_product_transfer"],
+    "author": "Camptocamp, BCIM, Akretion, Odoo Community Association (OCA)",
```
Comment (simahawk) [inline shopfloor_single_product_transfer_mobile/__manifest__.py]:
```suggestion
    "author": "Camptocamp, Odoo Community Association (OCA)",
```

File: shopfloor_single_product_transfer_mobile/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "maintainers": ["simahawk"],
```
Comment (simahawk) [inline shopfloor_single_product_transfer_mobile/__manifest__.py]:
```suggestion
    "maintainers": ["yourself-if-you-want"],
```

**Review by mmequignon — APPROVED** [review]

File: shopfloor_single_product_transfer_mobile/static/src/scenario/single_product_transfer.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                return null;
+            }
+            return {
+                record: data.move_line.product,
+                identifier: data.move_line.product.name,
+            };
+        },
+        product_detail_options_for_set_quantity: function () {
+            return {
+                title_action_field: {action_val_path: "product.display_name"},
```
Comment (TDu) [inline shopfloor_single_product_transfer_mobile/static/src/scenario/single_product_transfer.js]:
```suggestion
                title_action_field: {action_val_path: "product.barcode"},
```
Or the click on the info button that calls the scan_anything endpoint will not work.

**Comment by JuMiSanAr** [comment]
The README for shopfloor has been updated to reflect the need to add hooks if new scenarios are added:
https://github.com/OCA/wms/pull/641

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@JuMiSanAr can you rebase?

**Comment by JuMiSanAr** [comment]
> @JuMiSanAr can you rebase?

@simahawk Done!

**Pull request merged**